### PR TITLE
Adds loader icon to index.html along with the CSS that goes with it.

### PR DESCRIPTION
**Pull request by rcorral**
We want the application to feel as speedy as possible. To help with this, we can show the loading icon as fast as possible.
With this PR we add the icon and its accompanying CSS to the index.html file so that it can be shown immediately.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @kennyt

**Comment by kennyt** [comment]
tests failing

File: webpack/webpack.config.babel.js [hunk 1]
```diff
@@ -21,6 +66,26 @@ module.exports = {
...
+      // Replace HTML comments
+      {
+        test: /\.html$/,
+        exclude: /node_modules/,
+        loader: StringReplacePlugin.replace({
+          replacements: [
+            {
+              pattern: /<!--\sBOOTSTRAP-HTML\s-->/g,
+              replacement: function (match) {
+                return '<div id="canvas">' +
```
Comment (mlunoe) [inline webpack/webpack.config.babel.js]: #nit Could we make this like so to follow our style elsewhere and make it more readable?
```js
return (
  '<div id="canvas">' +
    '<div class="application-loading-indicator container ' +
      'container-pod vertical-center">' +
      bootstrap.HTML +
    '</div>' +
  '</div>'
);
```

File: webpack/webpack.config.babel.js [hunk 2]
```diff
@@ -45,6 +110,37 @@ module.exports = {
...
+                htmlContents = htmlContents
+                  .replace(/"\s+\+\s+require\(".*?"\)\s+\+\s+"/g, '');
+                // Load as if it were a module.
+                let compiledHTML = requireFromString(htmlContents);
+
+                let css = purifycss(compiledHTML, bootstrap.CSS, {
+                  minify: true
+                });
+
+                // Webpack doo doo's its pants with some of this CSS for
```
Comment (mlunoe) [inline webpack/webpack.config.babel.js]: :joy:

File: webpack/webpack.config.babel.js [hunk 3]
```diff
@@ -12,6 +21,42 @@ function absPath() {
 // Can override this with npm config set dcos-ui:external_plugins ../some/relative/path/to/repo
 let externalPluginsDir = absPath(process.env.npm_config_externalplugins || 'plugins');
 
+new Promise(function (resolve, reject) {
+  let cssEntryPoint = path.join(__dirname, '../src/styles/index.less');
+  less.render(fs.readFileSync(cssEntryPoint).toString(), {
+    filename: cssEntryPoint,
+    plugins: [colorLighten]
+  }, function (e, output) {
```
Comment (mlunoe) [inline webpack/webpack.config.babel.js]: #nit could we use `error`? could be mistaken with `event` or something else

File: webpack/webpack.config.babel.js [hunk 4]
```diff
@@ -12,6 +21,42 @@ function absPath() {
...
+    })
+    .catch(reject);
+  });
+}).then(function (css) {
+  bootstrap.CSS = css;
+});
+
+function requireFromString(src, filename) {
+  var Module = module.constructor;
+  var m = new Module();
```
Comment (mlunoe) [inline webpack/webpack.config.babel.js]: #nit `sourceModule`?

File: webpack/webpack.config.babel.js [hunk 5]
```diff
@@ -21,6 +66,26 @@ module.exports = {
...
     preLoaders: [
+      // Replace HTML comments
+      {
+        test: /\.html$/,
+        exclude: /node_modules/,
+        loader: StringReplacePlugin.replace({
+          replacements: [
+            {
+              pattern: /<!--\sBOOTSTRAP-HTML\s-->/g,
+              replacement: function (match) {
```
Comment (mlunoe) [inline webpack/webpack.config.babel.js]:
```js

/jenkins/workspace/public-dcos-ui-pulls/webpack/webpack.config.babel.js
  77:38  error  'match' is defined but never used  no-unused-vars

��� 1 problem (1 error, 0 warnings)
```

That should be the final linting error. Build succeeds after that

**Comment by mlunoe** [comment]
Looks good! A few small stylish comments, but I'd ⛵ after that!

File: package.json [hunk 6]
```diff
@@ -85,7 +85,9 @@
     "localStorage": "1.0.3",
     "marked": "0.3.5",
     "node-libs-browser": "1.0.0",
+    "postcss": "5.0.21",
     "postcss-loader": "0.8.2",
+    "purify-css": "1.1.9",
```
Comment (kennyt) [inline package.json]: mom, I finally made it

File: webpack/webpack.config.babel.js [hunk 7]
```diff
@@ -12,6 +21,42 @@ function absPath() {
...
+      resolve(prefixed.css);
+    })
+    .catch(reject);
+  });
+}).then(function (css) {
+  bootstrap.CSS = css;
+});
+
+function requireFromString(src, filename) {
+  var Module = module.constructor;
```
Comment (kennyt) [inline webpack/webpack.config.babel.js]: `let`

File: webpack/webpack.config.babel.js (hunk 1 above)
Comment (rcorral) [inline webpack/webpack.config.babel.js]: Done.

File: webpack/webpack.config.babel.js (hunk 3 above)
Comment (rcorral) [inline webpack/webpack.config.babel.js]: Done.

File: webpack/webpack.config.babel.js (hunk 4 above)
Comment (rcorral) [inline webpack/webpack.config.babel.js]: Done.

File: webpack/webpack.config.babel.js (hunk 5 above)
Comment (rcorral) [inline webpack/webpack.config.babel.js]: Fixed

File: webpack/webpack.config.babel.js (hunk 7 above)
Comment (rcorral) [inline webpack/webpack.config.babel.js]: Done.

**Comment by kennyt** [comment]
don't think the CSS is being applied.
gif

**Comment by mlunoe** [comment]
I'm hitting the same issue

**Comment by rcorral** [comment]
Fixed it. See what happens when you make me rename things?

**Pull request merged**